### ISSUE TITLE
[BUG] minVersion returns null for valid ranges

### ISSUE THREAD

**Issue by alfredsgenkins** [npm/node-semver#340]
# What / Why

For the case of:
- `^2.16.2 ^2.16`
- `>=2.16.2 <3.0.0 >=2.16.0 <3.0.0`

Function `semver.minVersion` can not define a valid range. the return value is null. While, I expect it to return `2.16.2`.

## When

Node version is `v14.9.0` and `semver` version is `6.14.8`.  

## Where

The code snippet is:

```js
const semver = require('semver');
const versionsExpected = '^2.16.2 ^2.16';
console.log(semver.satisfies('2.16.2', versionsExpected)); // returns true
console.log(semver.minVersion(versionsExpected)); // returns null, expected 2.16.2
```

**Comment by alfredsgenkins** [comment]
I know @coreyfarrell implemented this in https://github.com/npm/node-semver/pull/241. Do you have any idea on where to start? I would like to contribute.

**Comment by alfredsgenkins** [comment]
Like I saw the PR changes, but the logic is not descriptive to me. Can you briefly describe what was the idea behind the calculation?

Maybe I am doing something wrong?

**Comment by lukekarrys** [comment]
Fixed by #341
